src/v/v8_engine/tests/executor_test.cc

```diff
@@ -0,0 +1,178 @@
+
+    for (int i = 0; i < 3; ++i) {
+        futures.emplace_back(test_executor.submit(
+          task_for_test(), std::chrono::milliseconds(5000)));
+    }
+
+    ss::when_all_succeed(futures.begin(), futures.end()).get();
+    test_executor.stop().get();
+}
+
+SEASTAR_THREAD_TEST_CASE(simple_stop_executor_test) {
+    struct task_for_test {
+        void operator()() {}
+
+        void cancel() {}
+
+        void on_timeout() {}
+    };
+
+    v8_engine::executor test_executor(1, ss::smp::count);
+    test_executor.stop().get();
+    auto fut = test_executor.submit(
+      task_for_test(), std::chrono::milliseconds(5000));
+
+    BOOST_REQUIRE_EXCEPTION(
+      fut.get(),
+      ss::gate_closed_exception,
+      [](const ss::gate_closed_exception& e) {
+          return "gate closed" == std::string(e.what());
+      });
+}
+
+SEASTAR_THREAD_TEST_CASE(task_with_exception_test) {
+    class test_exception final : public std::exception {
+    public:
+        explicit test_exception(ss::sstring msg) noexcept
+          : _msg(std::move(msg)) {}
+
+        const char* what() const noexcept final { return _msg.c_str(); }
+
+    private:
+        ss::sstring _msg;
+    };
+
+    struct task_for_test_with_process_exception {
+        void operator()() { throw test_exception("Process exception"); }
+
+        void cancel() {}
+
+        void on_timeout() {}
+    };
+
+    struct task_for_test_with_process_set_cancel_exception {
+        void operator()() { sleep(1); }
+
+        void cancel() { throw test_exception("Set cancel() exception"); }
+
+        void on_timeout() {}
+    };
+
+    struct task_for_test_with_process_set_on_timeout_exception {
+        void operator()() { sleep(1); }
+
+        void cancel() {}
+
+        void on_timeout() {
+            throw test_exception("Set on_timeout() exception");
+        }
+    };
+
+    v8_engine::executor test_executor(1, ss::smp::count);
+
+    auto fut_process = test_executor.submit(
+      task_for_test_with_process_exception(), std::chrono::milliseconds(20));
+
+    BOOST_REQUIRE_EXCEPTION(
+      fut_process.get(), test_exception, [](const test_exception& e) {
+          return "Process exception" == std::string(e.what());
+      });
+
+    auto fut_cancel = test_executor.submit(
+      task_for_test_with_process_set_cancel_exception(),
+      std::chrono::milliseconds(20));
+
+    BOOST_REQUIRE_EXCEPTION(
+      fut_cancel.get(), test_exception, [](const test_exception& e) {
+          return "Set cancel() exception" == std::string(e.what());
+      });
+
+    auto fut_on_timeout = test_executor.submit(
+      task_for_test_with_process_set_on_timeout_exception(),
+      std::chrono::milliseconds(20));
+
+    BOOST_REQUIRE_EXCEPTION(
+      fut_on_timeout.get(), test_exception, [](const test_exception& e) {
+          return "Set on_timeout() exception" == std::string(e.what());
+      });
+
+    test_executor.stop().get();
+}
+
+SEASTAR_THREAD_TEST_CASE(stop_executor_with_finish_item_test) {
```

Comment:
these tests are all really great. one thing I'd like to see (did I miss it?) is a test that puts more stress on the system. to me the test would ideally:

* have an endless number of seastar submitted tasks on an independent background fiber being submitted to the executor
* the functions submitted would be both blocking and non-blocking
* be verifiable: that is, the test should be able to sanity check them for some sort of correctness maybe they return their input plus 1 or something. there are tons of options here.
* and run for a fixed amount of time, say, 15 seconds.

then repeat that for a couple different queue lengths like 1, 10, 10000

Reply:
done